pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -341,6 +342,82 @@ func handleInitUpdateVerifiedObjects(ctx *verifierContext) {
 	log.Infoln("handleInitUpdateVerifiedObjects done")
 }
 
+func verifiedImageStatusFromImageFile(
+	objType, objDirname, parentDirname, imageFileName string,
+	size int64) types.VerifyImageStatus {
+
+	status := types.VerifyImageStatus{
+		ObjType:  objType,
+		State:    types.DOWNLOADED,
+		Size:     size,
+		RefCount: 0,
+		LastUse:  time.Now(),
+	}
+
+	if objType == "appImg.obj" {
+		// Currently, for App Images, there are two conventions of
+		// ImageNames / Directory structures.
+		//  Containers - verified/<image-ID>/sha-<SHA>.aci
+		//  VMs - verified/<SHA>/safename
+		imageID, err := uuid.FromString(parentDirname)
+
+		log.Debugf("verifiedImageStatusFromImageFile: objType: %s, "+
+			"objDirname: %s, parentDirname: %s, imageFileName: %s, "+
+			"imageID: %s", objType, objDirname, parentDirname,
+			imageFileName, imageID.String())
+
+		// XXX - We are deciding if the image is a container based
+		//  on the fact that VM has a different dir structure. In future,
+		//  VMs will also move to use the same structure. We need to
+		//  revisit this logic then.
+		if err == nil {
+			// Container image
+			status.IsContainer = true
+			if strings.HasSuffix(imageFileName, ".aci") == false {
+				// XXX - Should we delete this image and not recover it??
+				log.Warnf("verifiedImageStatusFromImageFile - Container Image"+
+					"Doesn't have .aci Suffix: objType: %s, "+
+					"objDirname: %s, parentDirname: %s, imageFileName: %s, "+
+					"imageID: %s", objType, objDirname, parentDirname,
+					imageFileName, imageID.String())
+			}
+			if strings.HasPrefix(imageFileName, "sha") == false {
+				// XXX - Should we delete this image and not recover it??
+				log.Warnf("verifiedImageStatusFromImageFile - Container Image"+
+					"Doesn't have prefix sha. objType: %s, "+
+					"objDirname: %s, parentDirname: %s, imageFileName: %s, "+
+					"imageID: %s", objType, objDirname, parentDirname,
+					imageFileName, imageID.String())
+			}
+			status.ContainerImageID = strings.TrimSuffix(imageFileName, ".aci")
+			status.ImageID = imageID
+			// For Containers, ImageSha256 is ImageID
+			status.ImageSha256 = status.ImageID.String()
+			// For Containers, ImageID is the Safename
+			status.Safename = status.ImageID.String()
+		} else {
+			// VM Image
+			// XXX - Combine both the Schemes.. VM should also follow
+			//  ImageID based naming.
+			status.ImageSha256 = parentDirname
+
+			// We don't know the URL; Pick a name which is unique
+			if status.ImageSha256 != "" {
+				status.Safename = imageFileName + "." + status.ImageSha256
```

Comment:
Why are you changing this? Old code was
			safename := location.Name() + "." + sha	
thus there is always a .sha at the end, and I suspect some other code assumes that.

Reply:
Hmm.. Not intentional.. I probably mis-read the current code.. Let me check..